visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+	<div class="t-tags-input max-w-max min-w-full min-h-[43px] relative">
+		<div class="font-bold tracking-widest pb-2" v-if="label">
+			{{ label.toUpperCase() }}
+		</div>
+		<base-multiple-value-input
+			v-if="isSingleTag"
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+			display-key="value"
+			value-key="value"
+		/>
+		<base-tags-input
+			v-else
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+		/>
+	</div>
+</template>
+
+<script>
+	export default {
+		props: {
+			label: {
+				type: String,
+				default: '',
+			},
+			tKeyColumn: {
+				type: String,
+				default: 'KEY',
+			},
+			tValueColumn: {
+				type: String,
+				default: 'VALUE',
+			},
+			canReloadOnOpen: {
+				type: Boolean,
+				default: false,
+			},
+			isSingleTag: {
+				type: Boolean,
+				default: false,
+			},
+			placeholder: {
+				type: String,
+				default: '',
+			}
+		},
+		data: () => ({
+			is_filter: true,
+			selectedTags: [],
+			initialized: false,
+			tempTags: [], // base-multiple-value-input does not show selected items if `tags` is empty because data is fetched on open.
+		}),
+		computed:{ 
+			tags() {
+				const items = [];
+
+				if (this.rows.length) {
+					for (let row of this.rows) {
+						const key = row[this.tKeyColumn];
+						const value = row[this.tValueColumn];
+						if (key && value) {
+							items.push({
+								key: key.value,
+								value: value.value,
+							})
+						}
+					}
+				}
+				return items;
+			},
+			firstKey() {
+				// Used for single tag
+				if (this.tags.length) {
+					return this.tags[0].key;
+				} else if (this.tempTags) {
+					return this.tempTags[0].key;
+				}
+				return null;
```

Comment:
I try to avoid this kind of `if else if` code structures, what do you think about this option:
```suggestion
				if (this.tags.length) {
					return this.tags[0].key;
				}
				
				if (this.tempTags) {
					return this.tempTags[0].key;
				}
				
				return null;
```

Reply:
Got it, thanks.